src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py

```diff
@@ -2284,11 +2284,42 @@ def test_aks_create_with_http_proxy_config(self, resource_group, resource_group_
         # use custom feature so it does not require subscription to regiter the feature
         create_cmd = 'aks create --resource-group={resource_group} --name={name} --http-proxy-config={http_proxy_path} ' \
                      '--ssh-key-value={ssh_key_value} -o json'
-        self.cmd(create_cmd, checks=[
-            self.check('provisioningState', 'Succeeded'),
-            self.check('httpProxyConfig', None),
+
+        show_cmd = 'aks show --resource-group={resource_group} --name={name}'
+
+        try:
+            self.cmd(create_cmd)
+        except:
+            # expect previous to fail
```

Comment:
I suppose you could append `--no-wait` to the `create_cmd` which would not poll for results after sending the request successfully. This won't affect you checking the mc recorded by rp via the show command.

**But** you cannot update this cluster since the previous creating operation is in progress. A workaournd would be in a separate test case create a normal cluster first and update that cluster with the `--http-proxy-config` and `--no-wait` options.

Anyway, it's a little strange to test a cluster that can't be successfully provisioned. Is there any proper way to mock a valid proxy?

Reply:
I scripted out deploying a real proxy with VM in 2nd subnet. I'll update PR test case to use this.

Reply:
might still be a bit annoying with certs for live runs

Reply:
@FumingZhang how's that? it's a bit involved, but this is a much more realistic flow. added some docs as well.